### PR TITLE
[NEX-6555] Make gem compatible with Ruby 3

### PR DESCRIPTION
**Pull request by natedemery**
Makes this fork of [smart_listing](https://github.com/Sology/smart_listing) compatible with Ruby 3. This is based on [this example PR](https://github.com/Sology/smart_listing/pull/231) in the original repo

### CODE REVIEW

File: lib/smart_listing.rb [hunk 1]
```diff
@@ -109,7 +109,7 @@ def setup params, cookies
       else
         # let's sort by all attributes
         #
-        @collection = @collection.order(sort_keys.collect{|s| "#{s[1]} #{@sort[s[0]]}" if @sort[s[0]]}.compact) if @sort && !@sort.empty?
+        @collection = @collection.order(sort_keys.collect{|s| Arel.sql("#{s[1]} #{@sort[s[0]]}") if @sort[s[0]]}.compact) if @sort && !@sort.empty?
```
Comment (CJStadler) [inline lib/smart_listing.rb:112]: This doesn't seem related to Ruby 3, do you know what it is for?

Comment (natedemery) [reply lib/smart_listing.rb:112]: Arel is part of activerecord. I'm not sure _exactly_ what this addition gets us, to be honest. This was part of the Ruby 3 compatibility PR on the original gem.

Comment (natedemery) [reply lib/smart_listing.rb:112]: [Here's an article](https://dev.to/ashawareb/arel-and-ruby-on-rails-58m3) with some more information. It looks like it was introduced in Ruby 3.

**Review by CJStadler — APPROVED** [review]
Looks good to me! Will you open a PR on the nextrequest repo to use this fork?

**Comment by natedemery** [comment]
@CJStadler that's next on the list!
